### ISSUE TITLE
[ISSUE] Issue with `databricks_sql_endpoint` resource

### ISSUE THREAD

**Issue by michaelvonderbecke** [databricks/terraform-provider-databricks#3204]
### Configuration
Any databricks_sql_endpoint resource updates the plan every single time a plan/apply is called to update the 'health'.  This started happening sometime after 1.31.0 but not sure which specific version.  We have half a dozen SQL endpoints created through each terraform deployment and every single time a plan or apply runs it shows all of them getting updated with "health = (known after apply)". It pollutes the plan and makes it very annoying to approve pull requests.

### Expected Behavior
terraform plan should never show any updates to anything that shouldn't/hasn't changed.

### Actual Behavior
Every single databricks_sql_endpoint resource shows a change to the 'health' attribute every time a plan or apply runs, regardless if nothing has changed (we can run 10 applies in a row with 10 seconds between each one and each of them show all SQL endpoint health attributes changing.

### Steps to Reproduce
1. define a databricks_sql_endpoint resource
2. run terraform apply
3. run terraform plan or terraform apply as many times as you want and watch the health attribute always update

### Terraform and provider versions
I've tried both 1.34.0 and 1.35.0 - the behavior does not occur on our previous version (1.31.0)
Terraform version is 1.6.5

### Is it a regression?
Yes, worked fine on 1.31.0 and all previous versions

### Debug Output
I cannot easily get this information, if it's really needed please comment, otherwise its going to take a fairly large amount of effort to produce it.

**Comment by edwardfeng-db** [comment]
@michaelvonderbecke Thanks for raising, we tried but couldn't reproduce this on our side. Can you please provide the logs so that we can take a closer look? Thank you! You can follow the guide [here](https://registry.terraform.io/providers/databricks/databricks/latest/docs/guides/troubleshooting) to get the log.

**Comment by michaelvonderbecke** [comment]
> @michaelvonderbecke Thanks for raising, we tried but couldn't reproduce this on our side. Can you please provide the logs so that we can take a closer look? Thank you! You can follow the guide [here](https://registry.terraform.io/providers/databricks/databricks/latest/docs/guides/troubleshooting) to get the log.

Hi Edward,

It's going to take me quite awhile to get this as our deployment will not work properly outside of a github workflow/self hosted runner and we don't support debug logging (or have any way to get the log file after it's created) through our existing workflows.  I can't run a TF apply/plan locally due to all of the github secrets required to execute.

As a holdover I can show you what every single plan/apply looks like through our deployments now:

```
# module.databricks_workspace_1.databricks_sql_endpoint.anomalo[0] will be updated in-place
  ~ resource "databricks_sql_endpoint" "anomalo" {
      + health                    = (known after apply)
        id                        = "bf35f188985af37f"
        name                      = "anomalo"
        # (15 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.databricks_workspace_1.databricks_sql_endpoint.dbt[0] will be updated in-place
  ~ resource "databricks_sql_endpoint" "dbt" {
      + health                    = (known after apply)
        id                        = "18b53eb6ccdf78e0"
        name                      = "dbt"
        # (15 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.databricks_workspace_1.databricks_sql_endpoint.general_purpose will be updated in-place
  ~ resource "databricks_sql_endpoint" "general_purpose" {
      + health                    = (known after apply)
        id                        = "a0e6fd18c4fa1f3d"
        name                      = "General Purpose"
        # (15 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.databricks_workspace_1.databricks_sql_endpoint.jobs will be updated in-place
  ~ resource "databricks_sql_endpoint" "jobs" {
      + health                    = (known after apply)
        id                        = "607bf9a708a182c7"
        name                      = "jobs"
        # (15 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # module.databricks_workspace_1.databricks_sql_endpoint.tableau[0] will be updated in-place
  ~ resource "databricks_sql_endpoint" "tableau" {
      + health                    = (known after apply)
        id                        = "7031371a95bf775b"
        name                      = "tableau"
        # (15 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

Every single time we run a plan or apply this shows in the plan - all applying of all resources "completes successfully" and then if we immediately run another plan or apply the same thing shows up.

**Comment by michaelvonderbecke** [comment]
@edwardfeng-db I did a quick test to increase Terraform version to latest (1.7.2) to see if it makes a difference - it doesn't, same plan/apply over and over.

**Comment by michaelvonderbecke** [comment]
@edwardfeng-db I've attached a terraform plan with debug enabled and a terraform apply with debug enabled.  At this point this is the 50th+ apply I've run without changing any of the terraform code where "health" is being updated.

[15_Workspace Terraform Plan.txt](https://github.com/databricks/terraform-provider-databricks/files/14188989/15_Workspace.Terraform.Plan.txt)
[11_Workspace Terraform Apply.txt](https://github.com/databricks/terraform-provider-databricks/files/14189005/11_Workspace.Terraform.Apply.txt)

**Comment by michaelvonderbecke** [comment]
One additional note, as part of my development this morning I added a new sql endpoint resource that was created today, and *that* sql warehouse does not appear to constantly be trying to update the health attribute every time terraform plan/apply runs.  Not sure why that is but maybe it has something to do with a warehouse created with an older version of the provider/API/whatever?

**Comment by michaelvonderbecke** [comment]
Sorry for the spam, one more update, I checked the terraform state against an older warehouse that has this issue and the new warehouse that doesn't have the issue.

In the terraform state, the new warehouse has a block like this:

```
    health                    = [
        {
            details        = ""
            failure_reason = []
            message        = ""
            status         = "HEALTHY"
            summary        = ""
        },
    ]
```

The old warehouse does not have this block.  So, terraform is trying to get this info and add it to the state every single time an apply runs but apparently is having issues actually adding it to the state (no errors anywhere, its just not being added).

This appears to be the only attribute missing from the older warehouses.  I hope there is a way to fix this without removing warehouses from the state and readding them, or deleting/recreating them, as either of these would take a very significant amount of effort (we have 50+ workspaces with 5+ sql warehouses with this issue in each one, and each of these workspaces are in a different AWS account where removing and then importing an object to the state would be a lot of effort)

**Comment by alexott** [comment]
@michaelvonderbecke yep, this issue exists only for old SQL endpoints... The `health` attribute was introduced after 1.31.0 as part of the Go SDK migration. We potentially have a fix, but it may require broader review as it may affect some other resources.

**Comment by michaelvonderbecke** [comment]
@alexott @edwardfeng-db I tried removing one of the endpoints from the state, and then importing it back in, to see if it would automatically pick up and add the health attribute but it doesn't, so it seems like the only way to fix this currently is to delete and recreate the SQL endpoint?  (Which isn't viable for us as it would require tons of external apps changing their URLs).

Is there an ETA on the fix?  Or any other way to fix the issue ourselves that doesn't involve deleting and recreating the endpoints?  We have probably 200+ endpoints that have this issue.

**Comment by mc-meta** [comment]
Hello,

we are hit as well by this issue and have provided a debug output in a case with databricks support.

Any help greatly appreciated.

Thanks!

**Comment by edwardfeng-db** [comment]
The fix PR https://github.com/databricks/terraform-provider-databricks/pull/3227 is merged, it will go out with the new release this week.
